### PR TITLE
Upgrade AWS SDK for WebIdentityTokenCredentialsProvider Support

### LINKED ISSUE #76: Add the support for the WebIdentityTokenCredentialsProvider to run in AWS EKS
Greetings,

This plugin does not run well in Amazon EKS at the moment because it uses an old version of the aws-sdk-java.

Based on https://github.com/aws/aws-sdk-java/issues/2136, the version we need is at least 1.11.717. From that version, the WebIdentityTokenCredentialsProvider credentials provider should be used over the EC2ContainerCredentialsProviderWrapper and the Kinesis input plugin should start running fine in an EKS cluster.

The fix is potentially simple:
- Update the version of com.amazonaws:aws-java-sdk-core
- Update the version of com.amazonaws:aws-java-sdk-sts
- Update the version of com.amazonaws:amazon-kinesis-client

Should I open a PR?

Regards,
Jean-Philippe

### PR DESCRIPTION
**Pull request by mike-stewart**
This upgrades AWS SDK dependencies to add WebIdentityTokenCredentialsProvider authentication support. This is required to deploy in Amazon EKS using IAM Roles for Service Accounts.

Fixes #76. 

Closes #77. Closes #80. This duplicates those PRs, just with the build now passing and slightly newer dependencies.

### CODE REVIEW

**Comment by mike-stewart** [comment]
@codekitchen @jsvd @karenzone @robbavey @danhermann Any possibility of releasing this? Sounds like it would unblock a lot of folks.

**Comment by carlosescura** [comment]
Maybe also consider using the latest version with a recent fix: https://github.com/aws/aws-sdk-java/pull/2121

**Comment by mike-stewart** [comment]
@carlosescura I updated the versions here again - can you confirm these are correct?

**Comment by carlosescura** [comment]
Yes! As long as you are >= rev 787 it's fine 👍

**Comment by JPLachance** [comment]
Are we ready to merge this PR?

**Comment by bskim45** [comment]
maybe @robbavey or @roaksoax can help us out?

**Review by nathanwong91 — APPROVED** [review]

**Comment by alialperak** [comment]
I think a lot of us waiting for this PR, can it be merged?

File: logstash-input-kinesis.gemspec
Comment (jsvd) [inline logstash-input-kinesis.gemspec:25]:
```suggestion
  spec.requirements << "jar 'com.amazonaws:aws-java-sdk-core', '1.11.1034'"
  spec.requirements << "jar 'com.amazonaws:aws-java-sdk-sts', '1.11.1034'"
```

**Review by jsvd — APPROVED** [review]
LGTM

**Comment by jsvd** [comment]
This change has been published with version [2.2.0](https://rubygems.org/gems/logstash-input-kinesis/versions/2.2.0-java), thanks for the contribution :)

**Pull request merged**
